### ISSUE TITLE
OAuth Google not working on Heroku with custom domain.

### ISSUE THREAD

**Issue by levibostian** [RocketChat/Rocket.Chat#4666]
Your Rocket.Chat version: 0.40.1

I have an instance of Rocket.Chat on Heroku. Let's say it is `https://rocket.herokuapp.com`. 
I have a custom domain setup through Heroku. Let's say it is `http://rocket.example.com`. 

I have Google OAuth login setup to Rocket.Chat. If I go to the Heroku base domain: `https://rocket.herokuapp.com` and try to login through Google, success. I login successfully. 

However, if I try and login through my custom domain `http://rocket.example.com`, I login to my Google account, I "Allow" my application through OAuth, then my Google login popup window turns into a blank white screen. Just sitting there. I expect it to close and then redirect Rocket.Chat to login me in. 

Because of this, my guess is that my Google OAuth callback URL is not correct. It is set for the Heroku domain: `https://rocket.herokuapp.com` but I opened up the website and tried to login from my custom domain: `http://rocket.example.com`. 

I go into Rocket.Chat's admin settings and try to change my Google OAuth's callback URL to my custom domain, but the box is greyed out and does not allow me to change the URL. It is set automatically to my Heroku domain name. 

I went into the mLab MongoDB database and tried to edit the Google OAuth callback manually. 

Before:

```
{
    "_id": "Accounts_OAuth_Google_callback_url",
    "type": "relativeUrl",
    "readonly": true,
    "force": true,
    "enableQuery": "{\"_id\":\"Accounts_OAuth_Google\",\"value\":true}",
    "group": "OAuth",
    "section": "Google",
    "packageValue": "_oauth/google",
    "valueSource": "packageValue",
    "hidden": false,
    "blocked": false,
    "sorter": 7,
    "i18nLabel": "Accounts_OAuth_Google_callback_url",
    "i18nDescription": "Accounts_OAuth_Google_callback_url_Description",
    "value": "_oauth/google",
    "ts": {
        "$date": "2016-10-11T18:09:52.734Z"
    },
    "_updatedAt": {
        "$date": "2016-10-11T18:09:52.734Z"
    },
    "createdAt": {
        "$date": "2016-10-11T18:09:52.731Z"
    }
}
```

After: 

```
{
    "_id": "Accounts_OAuth_Google_callback_url",
    "type": "string",
    "readonly": true,
    "force": true,
    "enableQuery": "{\"_id\":\"Accounts_OAuth_Google\",\"value\":true}",
    "group": "OAuth",
    "section": "Google",
    "packageValue": "_oauth/google",
    "valueSource": "packageValue",
    "hidden": false,
    "blocked": false,
    "sorter": 7,
    "i18nLabel": "Accounts_OAuth_Google_callback_url",
    "i18nDescription": "Accounts_OAuth_Google_callback_url_Description",
    "value": "http://rocket.example.com/_oauth/google",
    "ts": {
        "$date": "2016-10-11T18:09:52.734Z"
    },
    "_updatedAt": {
        "$date": "2016-10-11T18:09:52.734Z"
    },
    "createdAt": {
        "$date": "2016-10-11T18:09:52.731Z"
    }
}
```

(I changed `value` and `type`)

And it does not seem to have done anything at all. My guess is Rocket.Chat needs to restart? Maybe I did not edit it correctly?

Long story short, how do I get _OAuth login_ to work while _hosted on Heroku_ with a _custom domain_ through Heroku?

**Comment by antgel** [comment]
Closing as ancient and support-related.
